tests/testthat/test-labels.r

```diff
@@ -62,6 +62,22 @@ test_that("Labels from default stat mapping are overwritten by default labels",
   expect_equal(p$labels$colour, "drv")
 })
 
+test_that("alt text is returned", {
+  p <- ggplot(mtcars, aes(mpg, disp)) +
+    geom_point()
+  expect_equal(get_alt_text(p), "A ggplot using a point layer")
+  p1 <- p + geom_smooth() + geom_point()
+  expect_equal(get_alt_text(p1), "A ggplot using point and smooth layers")
+  p1 <- p + labs(title = "A plot title")
+  expect_equal(get_alt_text(p1), "A ggplot titled 'A plot title' using a point layer")
+  p1 <- p + labs(subtitle = "A plot subtitle")
+  expect_equal(get_alt_text(p1), "A ggplot subtitled 'A plot subtitle' using a point layer")
+  p1 <- p + labs(title = "A plot title", subtitle = "A plot subtitle")
+  expect_equal(get_alt_text(p1), "A ggplot titled 'A plot title' and  subtitled 'A plot subtitle' using a point layer")
```

Comment:
I think I'd write this as: "A plot title: a plot subtitle. A plot made by ggplot2, displaying data using points, lines, and areas.`

Should we include x and y axis labels as well?

Reply:
So we shouldn't qualify that "A plot title" is the title?

Reply:
I don't think so? I think it'd usually be obvious from the context.

Reply:
I trust you on that

Reply:
As for x and y labels they are not necessarily materialised before rendering so adding those will require a bit more work that I think is not warranted (at least at this stage)

If you feel heavily for it we can do it of course